src/marqo/tensor_search/tensor_search.py

```diff
@@ -838,7 +851,10 @@ def get_query_vectors_from_jobs(
 
         if isinstance(q.q, dict) or q.q is None:
             ordered_queries = list(q.q.items()) if isinstance(q.q, dict) else None
-            weighted_vectors = []
+            # Store weights and vectors separately for use in interpolation
+            collected_weights: List[List[float]] = []
```

Comment:
nit: you can still use np to finish the computation inside this method and convert it to list[float] at the end.